src/main/java/com/epam/ta/reportportal/core/events/handler/LaunchFinishedEventHandler.java

```diff
@@ -127,16 +127,18 @@ public void onApplicationEvent(LaunchFinishedEvent event) throws ExecutionExcept
 				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, launch.getProjectId()));
 
 		AnalyzerConfig analyzerConfig = AnalyzerUtils.getAnalyzerConfig(project);
-		logIndexer.indexLaunchLogs(project.getId(), launch.getId(), analyzerConfig).get();
 
 		boolean isNotificationsEnabled = BooleanUtils.toBoolean(ProjectUtils.getConfigParameters(project.getProjectAttributes())
 				.get(ProjectAttributeEnum.NOTIFICATIONS_ENABLED.getAttribute()));
 
 		if (BooleanUtils.isTrue(analyzerConfig.getIsAutoAnalyzerEnabled()) && analyzerServiceAsync.hasAnalyzers()) {
 			List<Long> itemIds = analyzeCollectorFactory.getCollector(AnalyzeItemsMode.TO_INVESTIGATE)
 					.collectItems(project.getId(), launch.getId(), null);
-			analyzerServiceAsync.analyze(launch, itemIds, analyzerConfig)
-					.thenApply(it -> logIndexer.indexItemsLogs(project.getId(), launch.getId(), itemIds, analyzerConfig));
+			logIndexer.indexLaunchLogs(project.getId(), launch.getId(), analyzerConfig).join();
+			analyzerServiceAsync.analyze(launch, itemIds, analyzerConfig).join();
+			CompletableFuture.supplyAsync(() -> logIndexer.indexItemsLogs(project.getId(), launch.getId(), itemIds, analyzerConfig));
+		} else {
+			logIndexer.indexLaunchLogs(project.getId(), launch.getId(), analyzerConfig);
```

Comment:
check the if condition
you will start indexing even you don't have any analyzers

Reply:
logs should be indexed in any case. It doesn't depend on auto analysis